digitaltwin_client/src/dt_client_core.c

```diff
@@ -849,21 +724,30 @@ DIGITALTWIN_CLIENT_RESULT DT_ClientCoreRegisterInterfacesAsync(DT_CLIENT_CORE_HA
     {
         LogError("DT_InterfaceList_BindInterfaces failed, result = %d", result);
     }
-    else if ((result = SendInterfacesToRegisterMessage(dtClientCore, deviceCapabilityModel)) != DIGITALTWIN_CLIENT_OK)
+    else if (InvokeBindingSetDeviceMethodCallbackIfNeeded(dtClientCore, DTDeviceMethod_Callback) != 0)
+    {
+        // We start listening for incoming commands and properties after binding is invoked.
+        // If either of these fail, we will report an error to the application.
+        LogError("Cannot bind to device method callbacks");
```

Comment:
>callbacks [](start = 47, length = 9)

Nit: Please include the result in the LogError() message, and for the SetDeviceTwinCallbackIfNeeded() call too.

This will help make it easier to diagnose errors from logs only.